pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -100,7 +103,20 @@ def __init__(
         self._interaction = "XY" if self._seq._in_xy else "ising"
         self._qdict = self._seq.qubit_info
         self._size = len(self._qdict)
-        self._tot_duration = self._seq.get_duration()
+        self._modulated = bool(with_modulation)
+        if self._modulated and sequence._slm_mask_targets:
+            raise NotImplementedError(
+                "Simulation of sequences combining an SLM mask and output "
+                "modulation is not supported."
+            )
+        self._tot_duration = self._seq.get_duration(
+            include_fall_time=self._modulated
+        )
+        self.samples_obj = sampler.sample(
+            self._seq,
+            modulation=self._modulated,
+            extended_duration=self._tot_duration + 1,
```

Comment:
This is the eternal discussion on the ODE solver convergence issue which was fixed by adding one extra sample at the end. You had previously addressed this by adding the `additional_final_sample: bool = False` option.

Reply:
The usual suspect, alright.
Maybe add a comment so that we know where this pesky sample is introduced, I am sure it's far from the last time we deal with it.